libs/wss/src/wav.rs

```diff
@@ -0,0 +1,72 @@
+use std::io::Read;
+
+use crate::{Compression, Error, Wss};
+
+impl Wss {
+    /// Converts the WSS to a WAV file.
+    ///
+    /// # Errors
+    /// [`Error::Wav`] if an error occurs while writing the WAV file.
+    pub fn to_wav(&self) -> Result<Vec<u8>, Error> {
+        let mut cursor = std::io::Cursor::new(Vec::new());
+
+        let spec = hound::WavSpec {
+            channels: self.channels,
+            sample_rate: self.sample_rate,
+            bits_per_sample: self.bits_per_second,
+            sample_format: hound::SampleFormat::Int,
+        };
+        let mut writer = hound::WavWriter::new(&mut cursor, spec)?;
+
+        let samples = self.compression().decompress(&self.data);
+
+        for sample in samples {
+            writer.write_sample(sample)?;
+        }
+
+        drop(writer);
+
+        Ok(cursor.into_inner())
+    }
+
+    /// Create a new WSS file from a WAV file.
+    ///
+    /// # Errors
+    /// [`Error::Wav`] if an error occurs while reading the WAV file.
+    pub fn from_wav<R: Read>(wav: R) -> Result<Self, Error> {
+        Self::from_wav_with_compression(wav, &Compression::default())
+    }
+
+    /// Create a new WSS file from a WAV file with a specific compression type.
```

Comment:
Compression types have preconditions, I can't see checks for them?
bits per sample must be 16, channels must be 1 or 2.